### PR TITLE
quattor: functions : network: add subnet_prefix_to_mask

### PR DESCRIPTION
**Pull request by stdweird**
Add two new functions to convert a prefix length to netmask and vice versa.

### CODE REVIEW

File: quattor/functions/network.pan [hunk 1]
```diff
@@ -214,3 +214,41 @@ function get_subnet_params = {
...
     error(format("No subnet matching address %s found", ipaddr));
 };
+
+@documentation{
+    descr = Convert a network prefix (0..32) into a netmask (IPv4 notation), e.g. 16 becomes 255.255.0.0
+    arg = prefix
+}
+function subnet_prefix_to_mask = {
+    pref = ARGV[0];
+    if (!(is_long(pref) && pref >= 0 && pref <= 32)) {
```
Comment (jrha) [inline quattor/functions/network.pan]: There is a function for this `is_ipv4_prefix_length`

Comment (jrha) [reply quattor/functions/network.pan]: @stdweird any objection to using the existing function?

Comment (stdweird) [reply quattor/functions/network.pan]: done

**Comment by stdweird** [comment]
@jouvin travis fails because the SCDB test repo tool fails because some external svn repo needs a new pan compiler version

File: quattor/functions/network.pan [hunk 2]
```diff
@@ -214,3 +214,41 @@ function get_subnet_params = {
...
 };
+
+@documentation{
+    descr = Convert a network prefix (0..32) into a netmask (IPv4 notation), e.g. 16 becomes 255.255.0.0
+    arg = prefix
+}
+function subnet_prefix_to_mask = {
+    pref = ARGV[0];
+    if (!(is_long(pref) && is_ipv4_prefix_length(pref))) {
+        error("%s: argument %s must be a valid prefix", FUNCTION, pref);
```
Comment (jrha) [inline quattor/functions/network.pan]: This line needs the new panc version.

File: quattor/functions/network.pan [hunk 3]
```diff
@@ -214,3 +214,41 @@ function get_subnet_params = {
...
+    long_to_ip4(mask_long);
+};
+
+@documentation{
+    descr = Convert a netmask (IPv4 notation) into a network prefix (0..32), e.g. 255.255.0.0 becomes 16
+    arg = netmask
+}
+function subnet_mask_to_prefix = {
+    if (!is_ipv4(ARGV[0])) {
+        error("%s: argument %s must be a valid netmask", FUNCTION, ARGV[0]);
```
Comment (jrha) [inline quattor/functions/network.pan]: This line needs the new panc version.

File: quattor/functions/network.pan [hunk 4]
```diff
@@ -214,3 +214,41 @@ function get_subnet_params = {
...
+    };
+    masks = ip4_to_long(ARGV[0]);
+    mask = masks[0];
+    for (idx = 31; idx >= 0; idx = idx - 1) {
+        mask = mask / 2;
+        if (mask % 2 > 0) {
+            return(idx);
+        };
+    };
+    error("%s failed to determine prefix for mask %s %s", FUNCTION, ARGV[0]);
```
Comment (jrha) [inline quattor/functions/network.pan]: This line needs the new panc version.

**Comment by jrha** [comment]
Specifically, this change https://github.com/quattor/pan/pull/122, which implies panc 10.5.

**Comment by jouvin** [comment]
Test failure solved after fixing https://github.com/quattor/scdb/pull/45.

**Pull request merged**